### DISCUSSION TITLE
arm compile error with ndk android-toolchain

### DISCUSSION THREAD

**Discussion by bryanlinnan** [apache/incubator-mxnet#13610] [Q&A]
## Description
compile error occured when i tried to use android-toolchain to compile mxnet/amalgamation

## Environment info (Required)
ubuntu16.04  mxnet1.4 android-ndk-r14b/13b/16b

What to do:
1. generate android-toolchain
cd  ~/android-ndk-r14b/build/tools
./make-standalone-toolchain.sh --platform=android-21 --use-llvm --arch=arm --install-dir=~/Tool/android-toolchain

2. compile OpenBLAS
git clone https://github.com/xianyi/OpenBLAS.git
export PATH=$PATH:~/Tool/android-toolchain/bin 
cd OpenBLAS 
make TARGET=ARMV7 HOSTCC=gcc CC=arm-linux-androideabi-gcc NOFORTRAN=1
make PREFIX=~/Tool/openblas install

3. build mxnet
git clone --recursive https://github.com/dmlc/mxnet.git 
cd mxnet/amalgamation
export PATH=~/Tool/android-toolchain/bin:$PATH
export CXX=arm-linux-androideabi-g++
export CC=arm-linux-androideabi-gcc
make ANDROID=1

MXNet commit hash:
(Paste the output of `git rev-parse HEAD` here.)
f2ca66f2c537783aa60251080582793f42f395a7

## Error Message:
1. fatal error: sys/fcntl.h: No such file or directory
then i fix this error by modify the mxnet_predict-all.cc with fcntl instead of sys/fcntl
then another error come:
2. fatal error: x86intrin.h: No such file or directory
then i fix this error by removing it from mxnet_predict-all.cc
then another error come:
3.  error: 'stod' is not a member of 'std'
     error: 'to_string' is not a member of 'std'
now i don't know how to fix it

**Comment by vdantu** [comment]
@mxnet-label-bot add [build, question]
